lib/flight_scheduler/allocation_registry.rb

```diff
@@ -201,13 +208,21 @@ def load
     data = persistence.load
     return if data.nil?
     @lock.with_write_lock do
-      data.each do |hash|
-        allocation = Allocation.from_serialized_hash(hash)
-        if allocation
-          allocation.nodes(add: true).each do |node|
-            @node_allocations[node.name] << allocation
-          end
-          @job_allocations[allocation.job.id] = allocation
+      allocations = data.map { |h| [h['job_id'], Allocation.from_serialized_hash(h)] }
+      bad, good = allocations.partition { |_, a| a.job.nil? }
```

Comment:
As you've added validation, should we not be partitioning on the allocation being valid rather than just the case of the job being nil?

Alternatively, we could blindly add the allocations to the registry and catch the `AllocationConflict`.  Perhaps its better to ask for forgiveness here rather than permission.

Reply:
https://github.com/openflighthpc/flight-scheduler-controller/pull/48/commits/a3923a7edde5b6eb97427e23705cba7819f65344